### PR TITLE
LABS-443 added support for start and stop stream commands

### PR DESCRIPTION
**Pull request by steve-1820**

### CODE REVIEW

File: src/stream/commands/start_stream.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    bool adminOnly() const override {
+        return false;
+    }
+    std::string help() const override {
+        return "Registers an endpoint to the endpoint registry";
+    }
+
+public:
+    using Request = StartStream;
+    using Reply = StartStreamReply;
```
Comment (Z-Shang) [inline src/stream/commands/start_stream.cpp:57]: what are these type aliases used for?

File: src/stream/commands/start_stream.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+            LOGV2(999999, "Start stream command");
+
+            const auto nss = request().getNamespace();
+
+            StreamRegistry::get()->startStream(opCtx, nss.coll());
+
+            reply.setOk(true);
+
+            return reply;
```
Comment (michaelcahill) [inline src/stream/commands/start_stream.cpp:75]: Do we need all the blank lines?

**Review by michaelcahill — APPROVED** [review]
lgtm, thanks @steve-1820.

**Review by Z-Shang — APPROVED** [review]
lgtm
1 question
where are kCheckpoint / kRestore triggered?

File: src/stream/commands/start_stream.cpp (hunk 2 above)
Comment (steve-1820) [reply src/stream/commands/start_stream.cpp:75]: @michaelcahill I'll remove them

File: src/stream/commands/start_stream.cpp (hunk 1 above)
Comment (steve-1820) [reply src/stream/commands/start_stream.cpp:57]: There's documentation in commands.h from line 990. I'll link you in Slack.

**Pull request merged**